From: Georgy Kirichenko <kirichenkoga@gmail.com>
Date: Fri, 04 Dec 2015 19:00:36 +0000
Subject: [createaccount] Fix wrongly set username on a service instead of account
X-Git-Url: http://quickgit.kde.org/?p=kaccounts-integration.git&a=commitdiff&h=fe6b9c7b2511ba5ebc15542b0b7280c55e17ffa4
---
[createaccount] Fix wrongly set username on a service instead of account

BUG: 354949
FIXED-IN: 15.12.0
---


--- a/src/jobs/createaccount.cpp
+++ b/src/jobs/createaccount.cpp
@@ -141,8 +141,8 @@
 
     m_done = true;
 
+    connect(m_identity, &SignOn::Identity::credentialsStored, m_identity, &SignOn::Identity::queryInfo);
     m_identity->storeCredentials();
-    connect(m_identity, &SignOn::Identity::credentialsStored, m_identity, &SignOn::Identity::queryInfo);
 }
 
 void CreateAccount::pluginError(const QString &error)
@@ -177,12 +177,12 @@
         return;
     }
 
+    m_account->selectService();
+
     if (m_account->displayName().isEmpty()) {
         m_account->setDisplayName(info.userName());
     }
     m_account->setValue("username", info.userName());
-
-    m_account->selectService();
     m_account->setCredentialsId(info.id());
 
     Accounts::AuthData authData = m_accInfo->authData();
